docs/aws/audit/redshiftmonitoring/rules/redshift_desired_node_type.mdx

```diff
@@ -24,6 +24,223 @@ CBP
+4. For each cluster, check the "Node Type" column. This will display the type of nodes that the cluster is using. Compare this to your desired node type to see if there is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can download it from the official AWS website and configure it using the "aws configure" command. You will be prompted to provide your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. Once the AWS CLI is set up, you can list all the Redshift clusters in your account using the following command:
+
+   ```
+   aws redshift describe-clusters --query 'Clusters[*].[ClusterIdentifier]' --output text
+   ```
+
+   This command will return a list of all Redshift cluster identifiers.
+
+3. Now, for each cluster identifier, you can describe the cluster to get the node type. Use the following command:
+
+   ```
+   aws redshift describe-clusters --cluster-identifier <cluster-identifier> --query 'Clusters[*].[NodeType]' --output text
+   ```
+
+   Replace `<cluster-identifier>` with the actual cluster identifier. This command will return the node type of the specified Redshift cluster.
+
+4. Finally, compare the returned node type with your desired node type. If they don't match, then the Redshift cluster node is not of the desired type. You can automate this process using a Python script or a shell script to loop through all the clusters and compare their node types with the desired node type.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. This SDK allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc.
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create an AWS session using your credentials.
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2'
+)
+```
+
+3. Create a Redshift client using the session.
+
+```python
+redshift = session.client('redshift')
+```
+
+4. Use the `describe_clusters` method to retrieve information about all the Redshift clusters. Then, iterate over each cluster and check the node type.
+
+```python
+response = redshift.describe_clusters()
+
+for cluster in response['Clusters']:
+    if cluster['NodeType'] != 'Desired_Node_Type':
+        print(f"Cluster {cluster['ClusterIdentifier']} is not of the desired type.")
+```
+
+Replace 'Desired_Node_Type' with the node type you want your Redshift clusters to be. If a cluster is not of the desired type, the script will print its identifier.
+
+Remember to replace 'YOUR_ACCESS_KEY', 'YOUR_SECRET_KEY', and 'Desired_Node_Type' with your actual AWS access key, secret key, and desired Redshift node type, respectively.
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>